docs/pages/machine-id/troubleshooting.mdx

```diff
@@ -0,0 +1,286 @@
+---
+
+## A bot failed to renew a certificate due to a "generation mismatch", what does this mean?
+
+### Symptoms
+
+The bot will log an error like this:
+
+```text
+ERROR: renewable cert generation mismatch: stored=3, presented=2
+```
+
+Subsequent connection attempts by the bot may see errors like the following:
+```text
+ERROR: failed direct dial to auth server: auth API: access denied [00]
+"\tauth API: access denied [00], failed dial to auth server through reverse tunnel: Get \"https://teleport.cluster.local/v2/configuration/name\": Get \"https://example.com:3025/webapi/find\": x509: cannot validate certificate for example.com because it doesn't contain any IP SANs"
+"\tGet \"https://teleport.cluster.local/v2/configuration/name\": Get \"https://example.com:3025/webapi/find\": x509: cannot validate certificate for example.com because it doesn't contain any IP SANs"
+```
+
+In particular, note the message `auth API: access denied`. The Teleport auth
+server will provide some additional context:
+```text
+[AUTH]      WARN lock targeting User:"bot-example" is in force: The bot user "bot-example" has been locked due to a certificate generation mismatch, possibly indicating a stolen certificate. auth/apiserver.go:224
+```
+
+### Explanation
+
+Machine ID (with token-based joining) uses a certificate generation counter to
+detect potentially stolen renewable certificates. Each time a bot fetches a new
+renewable certificate, the counter is incremented and stored on the Teleport
+Auth Server and embedded in the certificate itself.
+
+If the counter embedded in your bot certificate doesn't match the counter
+stored in Teleport's Auth Server, the renewal will fail and the bot user will
+be automatically [locked](../access-controls/guides/locking.mdx). 
+
+Renewable certificates are exclusively stored in the bot's internal data
+directory, by default `/var/lib/teleport/bot`. It's possible to trigger this by
+accident if multiple bots are started using same internal data directory, or if
+this internal data is otherwise being shared between multiple bot instances.
+
+Additionally, if a bot fails to save its freshly renewed certificates (for
+example, due to a filesystem error) and crashes, it will attempt a renewal
+with old certificates and trigger a lock.
+
+### Resolution
+
+Before unlocking the bot, try to determine if either of the two scenarios
+described above apply. If the certificates were stolen, there may be 
+underlying security concerns that need to be addressed.
+
+Otherwise, first ensure only one bot instance is using the internal data
+directory. Multiple bots can be run on a single system, but separate data
+directories must be configured for each.
+
+Additionally, ensure the internal data is not being copied to any other nodes.
+If you'd like to share certificates between nodes, only copy content from
+destination directories.
+
+Once the underlying cause has been addressed, two steps required to reset a
```

Comment:
```suggestion
Once you have addressed the underlying cause, follow these steps to reset a
```
There's a lot of passive voice on this page, so I've suggested making this sentence more active.